components/eam/src/chemistry/modal_aero/modal_aero_amicphys.F90

```diff
@@ -3584,334 +3576,608 @@ end subroutine mam_soaexch_1subarea
-      real(r8), intent(inout), dimension( 1:max_aer, 1:max_mode ) :: &
-         qaer_cur
-      real(r8), intent(in   ), dimension( 1:max_aer, 1:max_mode ) :: &
-         qaer_del_grow4rnam
-      real(r8), intent(inout), dimension( 1:max_mode ) :: &
-         qwtr_cur
+      end subroutine mam_rename_1subarea
 
-      real(r8), intent(inout), optional, dimension( 1:max_mode ) :: &
-         qnumcw_cur
-      real(r8), intent(inout), optional, dimension( 1:max_aer, 1:max_mode ) :: &
-         qaercw_cur
-      real(r8), intent(in   ), optional, dimension( 1:max_aer, 1:max_mode ) :: &
-         qaercw_del_grow4rnam
-
-
-! !DESCRIPTION: 
-! computes TMR (tracer mixing ratio) tendencies for "mode renaming"
-!    during a continuous growth process
-! currently this transfers number and mass (and surface) from the aitken
-!    to accumulation mode after gas condensation or stratiform-cloud
-!    aqueous chemistry
-! (convective cloud aqueous chemistry not yet implemented)
-!
-! !REVISION HISTORY:
-!
+      !--------------------------------------------------------------------------------
+      !--------------------------------------------------------------------------------
 
-! local variables
-      integer :: iaer
-      integer :: mfrm, mtoo
-      integer :: n, npair
-
-      integer, parameter :: ldiag1 = 0
-
-      real(r8), parameter :: frelax = 27.0_r8
-      real(r8), parameter :: onethird = 1.0_r8/3.0_r8
-
-      real(r8) :: deldryvol_a(ntot_amode)
-      real(r8) :: deldryvol_c(ntot_amode)
-      real(r8) :: dp_belowcut(max_mode)
-      real(r8) :: dp_cut(max_mode)
-      real(r8) :: dgn_aftr, dgn_xfer
-      real(r8) :: dgn_t_new, dgn_t_old, dgn_t_oldaa
-      real(r8) :: dryvol_t_del, dryvol_t_new
-      real(r8) :: dryvol_t_old, dryvol_t_oldaa, dryvol_t_oldbnd
-      real(r8) :: dryvol_a(ntot_amode)
-      real(r8) :: dryvol_c(ntot_amode)
-      real(r8) :: dryvol_smallest(ntot_amode)
-      real(r8) :: factoraa(ntot_amode)
-      real(r8) :: factoryy(ntot_amode)
-      real(r8) :: lndp_cut(max_mode)
-      real(r8) :: lndgn_new, lndgn_old
-      real(r8) :: lndgv_new, lndgv_old
-      real(r8) :: num_t_old, num_t_oldbnd
-      real(r8) :: tailfr_volnew, tailfr_volold
-      real(r8) :: tailfr_numnew, tailfr_numold
-      real(r8) :: tmpa, tmpb, tmpd
-      real(r8) :: tmp_alnsg2(max_mode)
-      real(r8) :: v2nhirlx(ntot_amode), v2nlorlx(ntot_amode)
-      real(r8) :: xfercoef, xfertend
-      real(r8) :: xferfrac_vol, xferfrac_num, xferfrac_max
-      real(r8) :: yn_tail, yv_tail
+      subroutine find_renaming_pairs (nmodes, dest_mode_of_mode, &    !input
+           num_pairs, sz_factor, fmode_dist_tail_fac, v2n_lo_rlx, & !output
+           v2n_hi_rlx, ln_diameter_tail_fac, diameter_cutoff, &     !output
+           ln_dia_cutoff, diameter_threshold)    !output
 
+        !------------------------------------------------------------------------
+        !Find number of pairs which participates in the inter-mode transfer
```

Comment:
Can you explain what a pair is briefly in this comment?

Reply:
I moved the comments and added some more info.